pkg/controllers/tokentimeoutsync/tokentimeoutsync.go

```diff
@@ -0,0 +1,179 @@
+package tokentimeoutsync
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	"k8s.io/klog"
+
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	configinformer "github.com/openshift/client-go/config/informers/externalversions"
+	configv1lister "github.com/openshift/client-go/config/listers/config/v1"
+	oauthv1client "github.com/openshift/client-go/oauth/clientset/versioned/typed/oauth/v1"
+	oauthinformer "github.com/openshift/client-go/oauth/informers/externalversions"
+	oauthv1lister "github.com/openshift/client-go/oauth/listers/oauth/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+type tokenTimeoutSyncController struct {
+	oauthConfigLister configv1lister.OAuthLister
+	oauthClientLister oauthv1lister.OAuthClientLister
+	accessTokens      oauthv1client.OAuthAccessTokenInterface
+
+	globalTimeout  time.Duration
+	clientTimeouts map[string]*time.Duration
+}
+
+// knownConditionNames lists all condition types used by this controller.
+// These conditions are operated and defaulted by this controller.
+// Any new condition used by this controller sync() loop should be listed here.
+var knownConditionNames = sets.NewString(
+	"OAuthConfigGetFailed",
+	"OAuthClientListFailed",
+	"TokenListFailed",
+)
+
+func NewTokenTimeoutSyncController(
+	accessTokenClient oauthv1client.OAuthAccessTokenInterface,
+	oauthInformers oauthinformer.SharedInformerFactory,
+	configInformers configinformer.SharedInformerFactory,
+	recorder events.Recorder,
+) factory.Controller {
+	c := &tokenTimeoutSyncController{
+		accessTokens:      accessTokenClient,
+		oauthConfigLister: configInformers.Config().V1().OAuths().Lister(),
+		oauthClientLister: oauthInformers.Oauth().V1().OAuthClients().Lister(),
+
+		globalTimeout:  -1,
+		clientTimeouts: map[string]*time.Duration{},
+	}
+
+	return factory.New().ResyncEvery(30*time.Second).WithInformers(
```

Comment:
is 30s resync interval really common? pretty often for this kind of controller.

Reply:
@mfojtik told me we want it in case informers get stuck, I can live w/o it